x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -810,14 +843,23 @@ export class TimeSeriesExplorer extends React.Component {
     this.componentDidUpdate();
   }
 
-  componentDidUpdate(previousProps) {
+  componentDidUpdate(previousProps, previousState) {
     if (previousProps === undefined || previousProps.selectedJobId !== this.props.selectedJobId) {
       this.contextChartSelectedInitCallDone = false;
       this.setState({ fullRefresh: false, loading: true }, () => {
         this.loadForJobId(this.props.selectedJobId);
       });
     }
 
+    if (
+      previousProps === undefined ||
+      previousProps.selectedJobId !== this.props.selectedJobId ||
+      previousProps.selectedDetectorIndex !== this.props.selectedDetectorIndex ||
+      !isEqual(previousProps.selectedEntities, this.props.selectedEntities)
+    ) {
+      this.getFunctionDescription();
```

Comment:
Could we run into problems with a race condition since this is async function?
(Just trying to understand the implications here, since previously only relying on a change `previousProps` did make sure we check against from changes coming from outside the components - wondering if with including `perviousState` and then again triggering async state functions if we might run into render loops)